doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+ 
+   ***This command only work for trunk ports***
+
+  ```
+  
+![Vlan-MultipleAdd](https://user-images.githubusercontent.com/61490193/223199652-01b5ed35-15c0-480c-ad78-c78566db62df.png)
+
+
+ ####  Add all Vlan(s) as Vlan Member(s) excluding a range of vlans using a single command:
+
+  ```
+
+  admin@sonic:~$ sudo config vlan member add -m -e 12-17 PortChannel0003
+
+  Usage:  This command will add all existing vlan(s) execpt some specific as vlan member(s) 
+ 
+ Example : Suppose if Vlan10, Vlan11, Vlan12, Vlan13, Vlan14, Vlan15, Vlan16, Vlan17, Vlan18, Vlan19, Vlan20 are existing Vlans. This command will add PortChannel0003 as member of Vlan10, Vlan11, Vlan18, Vlan19, Vlan20
+
+    ***This command only work for trunk ports***
+
+  ```
+  
+  ![Portchannel-except](https://user-images.githubusercontent.com/61490193/223192175-396b487b-fe75-43f6-8598-d9e74773dd58.png)
+
+
+#### Multiple VLans command truncate:
+
+Multiple Vlan commands for adding Vlans (comma separated or in range) will truncate as soon as it finds any Vlan already exists even at the start or in the middle.  Deletion of Vlans works the same way and the command will truncate as soon as it finds a Vlan doesn’t exist. So, a user has to take care that a vlan should not already exist in the given range if he/she is trying to add multiple vlans.  Similarly, a user has to take care that all the vlans should exist in the given range if he/she is trying to delete multiple vlans. Following examples will make a more clear understanding of this fact.
+
+
+Example: Suppose Vlan 2 already exists then multiple range vlan command will truncate at the start and no vlan will be added as a result.
+
+![Vlan-Add-Error](https://user-images.githubusercontent.com/61490193/223192280-86424b5a-9cb5-4bea-94ea-c951522fb3a6.png)
+
+
+Example: Suppose Vlan 28 already exists then multiple range vlan command will truncate in the middle. Such that Vlan26 and Vla27 will be added and then the command will truncate with an error.
+
+![Vlan-Add-Middle](https://user-images.githubusercontent.com/61490193/223192341-7085c38d-421e-440c-8152-474b30a88d82.png)
+
+
+Example: Suppose Vlan 25 already exists then multiple range vlan command will truncate at the end.  Such that Vlan20 to Vlan24 will be added and then the command will truncate with an error.
+
+
+![Vlan-Add-Last](https://user-images.githubusercontent.com/61490193/223192431-d1a603a9-dcaf-42fb-bfad-23dde5e22437.png)
+
+We have given examples only for addition of vlans, commands for deletion of vlans will show similar behavior. 
+
+### Warm Boot and Fastboot Design Impact
+
+The existing warm boot/fast boot feature is not affected due to this design.
+
+## SAI API
+
+No SAI API change or addition is needed for this HLD. We will be using the existing SAI API.
+
+## Restrictions/Limitations
+
+This HLD did not consider hybrid switch port mode. We will be adding a separate HLD for hybrid mode. 
+
+### Testing Requirements/Design
+
```

Comment:
You should also mention a new show command that will display the siwtchport mode of the interface.  You should not be changing the output format of the existing "show interface status" output as it will break many existing applications, parsers, and testcases.  So it is best to disclose what is the new show command that you will be implementing here as well.

Also, I was looking for a section in this doc to clearly spell out all the possible transitions foe the switchport mode and what are allowed and not allowed.  Also, things as  if there are any default Access VLANs or Trunk untagged  (Native) VLAN when an interface is configured to switchport mode trunk.  It would be better to have a section dedicated to spell out all these for completeness and easy of review so that there are nothing unspecified and became ambiguous...

Reply:
We will add a new command for switchport modes and will not change existing "show int status" output format.
Also we will add more specific examples for Transition of port mode to avoid ambiguity.